tests/Middleware/RateLimiterTest.php

```diff
@@ -0,0 +1,161 @@
+<?php
+
+namespace Yiisoft\Yii\Web\Tests\Middleware;
+
+use Nyholm\Psr7\Factory\Psr17Factory;
+use Nyholm\Psr7\Response;
+use Nyholm\Psr7\ServerRequest;
+use PHPUnit\Framework\TestCase;
+use Psr\Http\Message\ResponseInterface;
+use Psr\Http\Message\ServerRequestInterface;
+use Psr\Http\Server\RequestHandlerInterface;
+use Psr\SimpleCache\CacheInterface;
+use Yiisoft\Cache\ArrayCache;
+use Yiisoft\Http\Method;
+use Yiisoft\Yii\Web\Middleware\RateLimiter;
+
+final class RateLimiterTest extends TestCase
+{
+    /**
+     * @test
+     */
+    public function singleRequestIsAllowed(): void
+    {
+        $middleware = $this->createRateLimiter($this->getCache());
+        $response = $middleware->process($this->createRequest(), $this->createRequestHandler());
+        $this->assertEquals(200, $response->getStatusCode());
+    }
+
+    /**
+     * @test
+     */
+    public function moreThanDefaultNumberOfRequestsIsNotAllowed(): void
+    {
+        $cache = $this->getCache();
+        $this->setRateLimiterCurrentRequestNumber($cache, 1000);
```

Comment:
While that is efficient, the test is knowing about cache key names. It might be worth it to call process N times instead. That would be closer to what happens in reality when requests are processed.

Reply:
fixed